Source/PlasticSourceControl/Private/PlasticSourceControlOperations.cpp

```diff
@@ -1810,6 +1821,94 @@ bool FPlasticShelveWorker::UpdateStates()
 }
 
 
+
+FName FPlasticUnshelveWorker::GetName() const
+{
+	return "Unshelve";
+}
+
+bool FPlasticUnshelveWorker::Execute(FPlasticSourceControlCommand& InCommand)
+{
+	TRACE_CPUPROFILER_EVENT_SCOPE(FPlasticUnshelveWorker::Execute);
+
+	// Get the state of the changelist to operate on
+	TSharedRef<FPlasticSourceControlChangelistState, ESPMode::ThreadSafe> ChangelistState = GetProvider().GetStateInternal(InCommand.Changelist);
+
+	// Detect if any file to unshelve has some local modification, which would fail the "unshelve" operation with a merge conflict
+	// NOTE: we could decide to automatically undo the local changes in order for this process too be automatic, like with Perforce
+	for (const FString& File : InCommand.Files)
+	{
+		if (ChangelistState->Files.FindByPredicate([&File](const FSourceControlStateRef& FileState) { return FileState->GetFilename().Equals(File, ESearchCase::IgnoreCase); }))
+		{
+			UE_LOG(LogSourceControl, Error, TEXT("Revert local changes in order to unshelve the corresponding changes from the shelve."));
```

Comment:
Wouldn't it be interesting from a user perspective to get the affected file path(s) logged here?
As the revert is only mandatory for conflictive files, the current message may look confusing.

Reply:
I created some code to create a string with the list of files, but it's too much code for not much. I ended up with just looping on the error message:

```
LogSourceControl: Error: Revert your changes to /Content/NewFolder/BP_CheckedOut.uasset in order to unshelve the corresponding changes from the shelve.
LogSourceControl: Error: Revert your changes to /Content/NewFolder/BP_Renamed.uasset in order to unshelve the corresponding changes from the shelve.
LogSourceControl: Error: Revert your changes to /Content/NewFolder/BP_New.uasset in order to unshelve the corresponding changes from the shelve.
```